contents/testing/security-testing.md

```diff
@@ -0,0 +1,144 @@
+
+Applications should have a documented design and architecture. This documentation can include models, textual documents, and other similar artifacts. It is essential to test these artifacts to ensure that the design and architecture enforce the appropriate level of security as defined in the requirements.
+
+Identifying security flaws in the design phase is not only one of the most cost-efficient places to identify flaws, but can be one of the most effective places to make changes. For example, if it is identified that the design calls for authorization decisions to be made in multiple places, it may be appropriate to consider a central authorization component. If the application is performing data validation at multiple places, it may be appropriate to develop a central validation framework (ie, fixing input validation in one place, rather than in hundreds of places, is far cheaper).
+
+### Step 3: During Development
+
+#### 3.1 Unit & System Tests
+
+System and unit tests should be written to explicitly ensure adherence to functional security requirements. For example, an application with an access control feature should include unit tests to ensure that unauthorised access credentials are not accepted.
+
+<box id="info">:fas-info-circle: If you would like to learn more about writing tests, you can read more [here](https://se-education.org/se-book/testing/).</box>
+
+#### 3.2 Code Walk Through
+
+The security team should perform a code walk through with the developers, and in some cases, the system architects. A code walk through is a high-level walk through of the code where the developers can explain the logic and flow of the implemented code. It allows the code review team to obtain a general understanding of the code, and allows the developers to explain why certain things were developed the way they were.
+
+The purpose is not to perform a code review, but to understand at a high level the flow, the layout, and the structure of the code that makes up the application.
+
+#### 3.3 Code Reviews
+
+Armed with a good understanding of how the code is structured and why certain things were coded the way they were, the tester can now examine the actual code for security defects.
+
+### Step 4: During Deployment‎
+
+#### 4.1 Penetration Testing
+
+Having tested the requirements, analyzed the design, and performed code review, it might be assumed that all issues have been caught. Hopefully this is the case, but penetration testing the application after it has been deployed provides a last check to ensure that nothing has been missed. 
+
+ZAP provides a beginner friendly introduction to penetration testing with a focus on web applications[^2].
+
+### At A Glance
+
+The following is the complete OWASP Testing Framework Workflow taken from the OWASP Testing Guide[^1]:
+
+<center>
+
+<pic src="security-testing/Sec-Test-Fig-2.png" alt="Workflow">
+Figure 2 - OWASP Testing Framework Workflow.
+</pic>
+</center>
+
+<br>
+<br>
+
```

Comment:
Perhaps the final section can have a `What's next` section if they want to learn more about security testing? How they can move forward and go deeper with security testing.

Reply:
Hmm I feel like the references are quite comprehensive, maybe an info box letting them know to check out the references if they want to learn more about security testing?

Reply:
Because I feel like readers usually won't really check out the references, as the references act as the support for your points, not really about what specifically to do next after reading the article.

Reply:
Hmm okay I get your point, I'll include a short section before the references :)